test/SqlExtension.Tests/Integration Tests/SqlOutputBindingIntegrationTests.cs

```diff
@@ -85,8 +87,23 @@ public void QueueTriggerProductsTest()
 
             Thread.Sleep(5000);
 
-            // Function should add 100 rows
-            Assert.Equal(100, this.ExecuteScalar("SELECT COUNT(1) FROM Products"));
+            // Function should add 100 rows (excluding ProductId >= 10000 which could be added by the timer trigger function)
+            Assert.Equal(100, this.ExecuteScalar("SELECT COUNT(1) FROM Products WHERE ProductId < 10000"));
+        }
+
+        [Fact]
+        public void TimerTriggerProductsTest()
+        {
+            // Since this function runs on a schedule (every 30 seconds), we don't need to invoke it.
+            // However the timer starts as soon as the Functions host starts, so we need to take that into account when calculating how long to wait.
+            // In total we will wait 60 seconds + 1 ms which should allow the function to run twice, generating 2000 rows of data.
```

Comment:
Can we guarantee that it'll only ever have 2000 rows though? If it starts as soon as the host starts then what if this test isn't ran until much later?

I think it would be fine to just refactor this test to wait for at least 2 ticks and then just verify that it has inserted at least 2 ticks worth of rows. (I don't think we should be checking they're exactly equal - since again that's non-deterministic depending on how fast the SELECT COUNT query executes)

Reply:
Changed it so we just check that data has been added. Also changed the Azure function to fire more frequently so we don't have to wait as long.